docs/source/common/production_inference.rst

```diff
@@ -1,48 +1,246 @@
+Convert to TorchScript
+======================
 
-TorchScript allows you to serialize your models in a way that it can be loaded in non-Python environments.
-The LightningModule has a handy method :meth:`~pytorch_lightning.core.lightning.LightningModule.to_torchscript`
-that returns a scripted module which you can save or directly use.
+`TorchScript <https://pytorch.org/docs/stable/jit.html>`_ allows you to serialize your models in a way that it can be loaded in non-Python environments.
+The ``LightningModule`` has a handy method :meth:`~pytorch_lightning.core.lightning.LightningModule.to_torchscript` that returns a scripted module which you
+can save or directly use.
 
 .. code-block:: python
 
+    class SimpleModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+
+        def forward(self, x):
+            return torch.relu(self.l1(x.view(x.size(0), -1)))
+
+
+    # create the model
     model = SimpleModel()
     script = model.to_torchscript()
 
     # save for use in production environment
     torch.jit.save(script, "model.pt")
 
 It is recommended that you install the latest supported version of PyTorch to use this feature without limitations.
+
+Once you have the exported model, you can run it in Pytorch or C++ runtime:
+
+.. code-block:: python
+
+    inp = torch.rand(1, 64)
+    scripted_module = torch.jit.load("model.pt")
+    output = scripted_module(dummy_input)
+
+------------
+
+PyTorch Runtime
+===============
+
+You can also load the saved checkpoint and use it as a regular :class:`torch.nn.Module`. You can extract all your :class:`torch.nn.Module`
+and load the weights using the checkpoint saved using LightningModule after training. For this, we recommend copying the exact implementation
+from your LightningModule ``init`` and ``forward`` method.
+
+.. code-block:: python
+
+    class Encoder(nn.Module):
+        ...
+
+
+    class Decoder(nn.Module):
+        ...
+
+
+    class AutoEncoder(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.encoder = Encoder()
+            self.decoder = Decoder()
+
+        def forward(self, x):
+            return self.encoder(x)
+
+        def training_step(self, batch, batch_idx):
+            x, y = batch
+            y_hat = self.encoder(x)
+            y_hat = self.decoder(y_hat)
+            loss = ...
+            return loss
+
+
+    class AutoEncoderProd(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.encoder = Encoder()
+            self.decoder = Decoder()
+
+        def forward(self, x):
+            return self.encoder(x)
+
+
+    # train it
+    trainer = Trainer(gpus=2)
+    trainer.fit(model, train_dataloader, val_dataloader)
+    trainer.save_checkpoint("best_model.ckpt")
+
+
+    # create the PyTorch model and load the checkpoint weights
+    model = AutoEncoderProd()
```

Comment:
To reload the model, we would need to drop the auto_encoder key.

Reply:
done!